src/inst.h

```diff
@@ -281,4 +322,35 @@ class IfInst : public FixedInputs<2>
     std::array<Inst *, 2> branchs_;
 };
 
+enum class ConditionCode {
+    NONE = 0,
+    EQ,
+    NE,
+    // Signed
+    GE,
+    GT,
+    LE,
+    LT,
+    // Unsigned NOT SUPPORTED now
+};
+
+class CompareInst : public FixedInputs<2>
```

Comment:
Has a small connection with the rest of the commit, should be placed in a separate commit